### DISCUSSION TITLE
Running RHVAE on one desktop computer

### DISCUSSION THREAD

**Discussion by axu-git** [clementchadebec/benchmark_VAE#128] [General]
I would like to get familiar with RHVAE by running a small test example on a desktop computer (i.e., no GPU). Is this possible?

I tried the MNIST example given in the google colab for RHVAE, reduced the test set to just 10 images (down from 50,000), 2 epochs, and it was still running 30 minutes later.

**Comment by clementchadebec** [comment]
I would not recommend doing this since this model is quite computationally demanding but with that number of images it should have definitively produced something. I just tested and in less than a minute, training was done. Did you use different parameters than the ones in the colab?

![Screenshot 2024-02-27 at 14 18 57](https://github.com/clementchadebec/benchmark_VAE/assets/47564971/07538e46-5cec-460a-8dd0-93e80f6c5706)

**Comment by axu-git** [comment]
I got as far as:

```
Preprocessing train data...
Checking train dataset...
Preprocessing eval data...

Checking eval dataset...
Using Base Trainer
```

And the output did not advance any further in the Jupyter notebook I was using. Have you had any experience with the code hanging at this particular step?

If not perhaps there is something wrong with my pip installation, so I should try to reinstall it/figure out what's wrong with it. The Jupyter notebook also froze (no error, just continually executing) when I tried to load the MNIST dataset from torchvision, but when I used keras to load the dataset that problem seemed to disappear. So the code that I used is exactly the same code as in the Google Colab document except I loaded from MNIST (chose a smaller number of training/eval datasets, and chose 2 epochs). Here's the code I used:

```
cwd = os.getcwd()
(x_train, y_train), (x_test, y_test) = keras.datasets.mnist.load_data(
                    path=cwd+"/mnist.npz")

x_train = x_train.astype("float32") / 255

train_dataset = x_train[0:10].reshape(-1, 1, 28, 28)
eval_dataset = x_train[10:15].reshape(-1, 1, 28, 28)

config = BaseTrainerConfig(
    output_dir='my_model',
    learning_rate=1e-4,
    per_device_train_batch_size=64,
    per_device_eval_batch_size=64,
    num_epochs=2,
)


model_config = RHVAEConfig(
    input_dim=(1, 28, 28),
    latent_dim=16,
    n_lf=1,
    eps_lf=0.001,
    beta_zero=0.3,
    temperature=1.5,
    regularization=0.001

)

model = RHVAE(
    model_config=model_config,
    encoder=Encoder_ResNet_VAE_MNIST(model_config), 
    decoder=Decoder_ResNet_AE_MNIST(model_config) 
)

pipeline = TrainingPipeline(
    training_config=config,
    model=model
)

pipeline(
    train_data=train_dataset,
    eval_data=eval_dataset
)
```

**Comment by axu-git** [comment]
Edit: everything works now, thanks for all your help! The problem was in some imports I used, and the code itself works perfectly.

**Reply by clementchadebec** [reply]
Cool! Happy to see that everything works well on your end as well :)
